lib/common/socketpool.c

```diff
@@ -56,7 +56,8 @@ struct st_h2o_socketpool_connect_request_t {
     h2o_socket_t *sock;
     h2o_multithread_receiver_t *getaddr_receiver;
     size_t selected_target;
-    size_t remaining_try_count;
+    uint32_t remaining_try_count;
+    uint32_t max_try_count;
```

Comment:
Do we need `max_try_count`? I think you might be able to obtain that from `h2o_socketpool_target_conf_t` when doing `max_try_count - remaining_try_count`. Though I might be wrong.

Regarding types, please refer to my comment above.

Reply:
How about 2612667  ?